nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/ops/SDNN.java

```diff
@@ -1032,4 +1033,35 @@ public SDVariable multiHeadDotProductAttention(String name, SDVariable queries,
             );
         }
     }
+
+    /**
+     * Max pooling on the input and outputs both max values and indices
+     *
+     * @param name  Name of the output variable
+     * @param x input array
+     * @return output array and argmax array
+     */
+    public SDVariable[] maxPoolWithArgmax(String name, SDVariable x, Pooling2DConfig pooling2DConfig) {
+        SDVariable[] res = f().maxPoolWithArgmaxs(x, pooling2DConfig);
+        return res;
+    }
+
+    /**
+     * Batch normalization
+     *
+     * @param name  Name of the output variable
+     * @param x 4D array
+     * @param scale vector for scaling factor of normalized x
+     * @param offset vector to shift to the normalized x
+     * @param dataFormat integer scalar - data format
+     * @param isTraining boolean scalar - is training mode
+     * @return y: 4D array
+     *         batch_mean: vector
+     *         batch_var: vector
+     */
+    public SDVariable[] fusedBatchNorm(String name, SDVariable x, SDVariable scale, SDVariable offset,
+                                       SDVariable dataFormat, SDVariable isTraining) {
+        SDVariable[] res = f().fusedBatchNorm(x,scale,offset,dataFormat,isTraining);
+        return res;
```

Comment:
Missing updateVariableNameAndReference, `@NonNull`. Name arg javadoc should mention it can be null.

Reply:
There are no @NonNull  annotations here, is it correct line of code?